### PR TITLE
Add rules to recommended config; Setup minimal test suite

### PR DESCRIPTION
**Pull request by aaronb97**
- Adds various rules to recommended config
- Sets up small test suite with vitest mainly to make sure the stylistic plugin is working as expected

### CODE REVIEW

File: lib/recommended.js [hunk 1]
```diff
@@ -73,10 +73,46 @@ module.exports = {
...
+            {
+                blankLine: "always",
+                prev: [
+                    "multiline-block-like",
+                    "multiline-expression",
+                    "multiline-const",
+                ],
+                next: "*",
+            },
+        ],
```
Comment (albatrocity) [inline lib/recommended.js]: Looks like this was [deprecated in in ESLint v8.53.0](https://eslint.org/docs/latest/rules/padding-line-between-statements) and moved to [Eslint Stylistic](https://eslint.style/rules/js/padding-line-between-statements). I like the idea of this rule though, should we look into adding [the  plugin](https://eslint.style/guide/getting-started)?

**Review by luiz-chagas — COMMENTED** [review]
I like the additions, if you can switch over to the stylistic plugin instead of the deprecated rule this is good to go

File: lib/recommended.js [hunk 2]
```diff
@@ -66,17 +68,56 @@ module.exports = {
...
+        "react/self-closing-comp": ["error"],
+        "react/jsx-no-useless-fragment": ["error", { allowExpressions: true }],
+        "react/jsx-key": ["error"],
+        "react/jsx-sort-props": [
+            "error",
+            {
+                callbacksLast: true,
+                shorthandFirst: true,
+                reservedFirst: true,
+                noSortAlphabetically: true,
```
Comment (aaronb97) [inline lib/recommended.js:97]: 👀 Not really a fan of alphabetical props, you run in to all sorts of situations where you have a component like
```
<Something
	title="title"
	description="description"
/>
```

But the alphabetizing wants the props to be
```
<Something
	description="description"
	title="title"
/>
```

Let me know what you opinion is on this one

Comment (albatrocity) [reply lib/recommended.js:97]: I second this. It's annoying when you're trying to group related props together and the linter redistributes them alphabetically.

**Review by luiz-chagas — APPROVED** [review]
Looks good to me! :shipit:

**Pull request merged**